src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -428,7 +428,9 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
       const modifiedQuery = applyFiltersFromURL(location, collectionStats)
 
       requestAnimationFrame(() => {
-        useCollectionFilters.setState(modifiedQuery as any)
+        if (modifiedQuery) {
+          useCollectionFilters.setState(modifiedQuery as any)
```

Comment:
an alternative: `useCollectionFilters.setState(modifiedQuery || initialCollectionFilterState)`

Reply:
Actually resetting the state completely isn't what we want it will wipe out the market count. Setting the state to undefined actually does not mutate the state at all so it's like no change happened